plugins/locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayer.java

```diff
@@ -99,15 +99,24 @@ private void addSources() {
     // Create an empty feature collection
     FeatureCollection emptyFeature = FeatureCollection.fromFeatures(new Feature[] {});
 
-    GeoJsonSource locationSource = new GeoJsonSource(LocationLayerConstants.LOCATION_SOURCE, emptyFeature);
-    mapboxMap.addSource(locationSource);
+    if (mapboxMap.getSource(LocationLayerConstants.LOCATION_SOURCE) == null) {
```

Comment:
Are these changes required by 5.2? If so, 👍

Reply:
yes, it seems so, will fix this up in a seperate PR